### PR TITLE
Separate coil configuration from data #83

### PR DESCRIPTION
**Pull request by yozik04**

### CODE REVIEW

File: nibe/connection/encoders.py [hunk 1]
```diff
@@ -38,46 +38,31 @@ class CoilDataEncoder:
...
-    def encode(self, coil: Coil):
-        value = coil.value
+    def encode(self, coil_data: CoilData) -> bytes:
         try:
-            assert value is not None, "Unable to encode None value"
-            if coil.has_mappings:
-                return self._pad(
-                    self._get_parser(coil), coil.get_reverse_mapping_for(value)
-                )
+            assert coil_data.is_valid, "Invalid coil data"
```
Comment (elupus) [inline nibe/connection/encoders.py:43]: Maybe we should change is_valid() to check() or verify() and have it pass along the exception. Then we retain the reason for the failure.

**Review by elupus — APPROVED** [review]
Nice!

File: nibe/connection/encoders.py (hunk 1 above)
Comment (yozik04) [reply nibe/connection/encoders.py:43]: Do you mean `verify()` should throw exception or just return it?

Comment (elupus) [reply nibe/connection/encoders.py:43]: Throw it was my intention.

File: nibe/connection/nibegw.py [hunk 2]
```diff
@@ -268,19 +266,62 @@ async def read_coil(self, coil: Coil, timeout: float = DEFAULT_TIMEOUT) -> Coil:
...
             except asyncio.TimeoutError:
                 raise CoilReadTimeoutException(
                     f"Timeout waiting for read response for {coil.name}"
                 )
-            finally:
-                del self._futures["read"]
+            except DecodeException as e:
+                raise CoilReadException(
+                    f"Failed decoding response for {coil.name}: {e}"
+                ) from e
```
Comment (yozik04) [inline nibe/connection/nibegw.py:277]: @elupus If I start capturing exceptions here then it takes `timeout` * 3 to return from the method. Maybe you know how to fix... Couple tests capture that.

Comment (yozik04) [reply nibe/connection/nibegw.py:277]: Ahh, we have that magic retry logic there.... Damn...

**Comment by elupus** [comment]
Sounds good.

**Comment by yozik04** [comment]
@elupus I have renamed some exceptions and added some doc strings. If it is fine now I can merge and make a release.

**Comment by elupus** [comment]
Looks good. I will try to update the hass client for this.

**Pull request merged**